### ISSUE TITLE
"Method returned null" when building with --target parse

### ISSUE THREAD

**Issue by reid47** [gracelang/minigrace#109]
After pulling the new changes this morning, it seems that `--target parse` no longer works. Running `make test` builds the compiler successfully with 153/153 tests passed, and `./minigrace example.grace` successfully compiles and runs. But `./minigrace --target parse example.grace` gives the following error:
```
Method returned null: Object.node on line 127. This is a bug in the compiler runtime or an external native module. Terminating the program.
```
This happens with any code in `example.grace`, including just `print "hello world"`.

**Comment by apblack** [comment]
On 15 Jun 2015, at 17:35, Reid Mitchell <notifications@github.com> wrote:

> After pulling the new changes this morning, it seems that --target parse no longer works. Running make test builds the compiler successfully with 153/153 tests passed, and ./minigrace example.grace successfully compiles and runs. But ./minigrace --target parse example.grace gives the following error:
> 
> Method returned null: Object.node on line 127. This is a bug in the compiler runtime or an external native module. Terminating the program.
> This happens with any code in example.grace, including just print "hello world".
> 
> —
> Reply to this email directly or view it on GitHub.
> 
Thanks!  I’ll not be able to look at this until wednesday.

There should be tests for these flags, of course.

	Andrew

**Comment by reid47** [comment]
If it helps, `target --processed-ast` works and includes the AST.

**Comment by apblack** [comment]
On 16 Jun 2015, at 10:17 , Reid Mitchell <notifications@github.com> wrote:

> If it helps, target --processed-ast works and includes the AST.
> 
> �
> Reply to this email directly or view it on GitHub.
> 


I too noticed this fact this morning.  It makes no sense, because --target processed-ast (and, of course, --target js) produces the same AST that --target parse sees failing, and then does more.  But I'm in a committee meeting all day, and can't look at it.

	Andrew

**Comment by apblack** [comment]
This is fixed in df1cb64b755bd7264716794be622d9ec474bf4ce.  The problem was an uninitialized variable.  It was initialized in a later pass, which is why --target symbol and --target processed-ast worked, but --target parse did not.

However, I'm leaving this issue open, because the deeper bug is that the compiler is reporting an internal error rather than an uninitialized variable exception.

**Comment by apblack** [comment]
The problem of unititialized variables not raising an excpetion is now a separate issue, #113
